javav2/usecases/topics_and_queues/src/main/java/com/example/sns/SNSWorkflow.java

```diff
@@ -0,0 +1,669 @@
+            System.exit(1);
+        }
+    }
+
+    public static void pubMessageFIFO(SnsClient snsClient, String message, String topicArn, String msgAttValue, String duplication) {
+        try {
+            PublishRequest request;
+            // Means the user did not choose to use a message attribute.
+            UUID uuid = UUID.randomUUID();
+            UUID uuid1 = UUID.randomUUID();
+            if (msgAttValue.isEmpty() ) {
+                if (duplication.compareTo("y") == 0) {
+                    request = PublishRequest.builder()
+                        .message(message)
+                        .messageDeduplicationId(contentId)
+                        .messageGroupId(uuid1.toString())
+                        .topicArn(topicArn)
+                        .build();
+                } else {
+                    request = PublishRequest.builder()
+                        .message(message)
+                        .messageDeduplicationId(uuid.toString())
+                        .messageGroupId(uuid1.toString())
+                        .topicArn(topicArn)
+                        .build();
+                }
+
+            } else {
+                Map<String, MessageAttributeValue> messageAttributes = new HashMap<>();
+                messageAttributes.put(msgAttValue, MessageAttributeValue.builder()
+                    .dataType("String")
+                    .stringValue("true")
+                    .build());
+
+                if (duplication.compareTo("y") == 0) {
+                    request = PublishRequest.builder()
+                        .message(message)
+                        .messageDeduplicationId(contentId)
+                        .messageGroupId(uuid1.toString())
+                        .topicArn(topicArn)
+                        .build();
+                } else {
+                    // Create a publish request with the message and attributes.
+                    request = PublishRequest.builder()
+                        .topicArn(topicArn)
+                        .message(message)
+                        .messageDeduplicationId(uuid.toString())
+                        .messageGroupId(uuid1.toString())
+                        .messageAttributes(messageAttributes)
+                        .build();
+                }
+            }
+
+            // Publish the message to the topic.
+            PublishResponse result = snsClient.publish(request);
+            System.out.println(result.messageId() + " Message sent. Status was " + result.sdkHttpResponse().statusCode());
+
+
+        } catch (SnsException e) {
+            System.err.println(e.awsErrorDetails().errorMessage());
+            System.exit(1);
+        }
+    }
+
+    // Subscribe to the SQS queue.
+    public static String subQueue(SnsClient snsClient, String topicArn, String queueArn, List<String> filterList) {
+        try {
+            SubscribeRequest request;
+            if (filterList.isEmpty()) {
```

Comment:
Added an SOS here for subscribing with a filter, "sns_Subscribe_Filtered" and "sns_Subscribe_Queue". Both can point to the same code.